### PR TITLE
Stats: Enable tab switching for unsupported fields of hourly views

### PR DESCRIPTION
**Pull request by dognose24**
Related to https://github.com/Automattic/red-team/issues/301.

## Proposed Changes

* Restore chart tab switching for tab `Visitors`, `Likes`, and `Comments`.
* Use `NULL` to recognize unsupported Stats visits fields for hourly views to apply alternative data.

## Why are these changes being made?

* Making unsupported hourly views tabs clickable with dedicated text in an empty state would communicate better to users. Context: https://github.com/Automattic/wp-calypso/pull/97060#issuecomment-2518645225.

## Testing Instructions

* Apply Diff `D167991-code` and sandbox `public-api.wordpress.com`.
* Navigate to Stats > Traffic with the feature flag: `?flags=stats/new-date-filtering`.
* Click on any day on the chart to navigate to the hourly views.
* Ensure the `Views` tab works as previously.

截圖 2024-12-06 下午2 58 22

* Ensure the `Visitors`, `Likes`, and `Comments` tabs are clickable.
* Click on the `Visitors` tab.
* Ensure the chart is showing empty state with hourly views dedicated texts.
* Ensure the chart tab still displays the cumulative number of visitors for a single day.

截圖 2024-12-06 下午2 58 05

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-tabs/index.jsx [hunk 1]
```diff
@@ -31,7 +31,8 @@ class StatsTabs extends Component {
 		} else {
 			data?.map( ( day ) =>
 				tabs.map( ( tab ) => {
-					if ( isFinite( day[ tab.attr ] ) ) {
+					// Exclude non-numeric values, e.g., NULL from unsupported stats fields.
+					if ( Number.isFinite( day[ tab.attr ] ) ) {
```
Comment (dognose24) [inline client/my-sites/stats/stats-tabs/index.jsx:35]:
```
isFinite( null ) === true
Number.isFinite( null ) === false
```

Comment (kangzj) [reply client/my-sites/stats/stats-tabs/index.jsx:35]: Hmmmm that's why we love JavaScript right? 🥲

**Review by kangzj — APPROVED** [review]
LGTM 🚀

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17054550

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @dognose24 for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17054550).

**Pull request merged**